pupil_src/shared_modules/surface_tracker/surface.py

```diff
@@ -538,8 +538,8 @@ def update_heatmap(self, gaze_on_surf):
             self.within_surface_heatmap[:, :, 3] = 125
         self.within_surface_heatmap[:, :, :3] = color_map
 
-    def get_uniform_heatmap(self):
-        hm = np.zeros((1, 1, 4), dtype=np.uint8)
+    def get_uniform_heatmap(self, resolution):
+        hm = np.zeros((*resolution, 4), dtype=np.uint8)
         hm[:, :, :3] = cv2.applyColorMap(hm[:, :, :3], cv2.COLORMAP_JET)
```

Comment:
This should ensure that resolution has the correct number of elements or fail right there.